File: examples-sdk/kotlin/009-basic-credit-regulatory-slik/main.kt
// 009-basic-credit-regulatory-slik — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 009-basic-credit-regulatory-slik/main.kt --release

fun main() {
    val p = VilPipeline("RegulatoryStreamPipeline", 3083)
    p.sink("regulatory_sink", 3083, "/regulatory-stream")
    p.source("regulatory_source", url = "http://localhost:18081/api/v1/credits/ndjson?count=1000", format = "json")
    p.route("sink.trigger_out", "source.trigger_in", "LoanWrite")
    p.route("source.response_data_out", "sink.response_data_in", "LoanWrite")
    p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")
    p.compile()
}

File: examples-sdk/swift/009-basic-credit-regulatory-slik/main.swift
// 009-basic-credit-regulatory-slik — Swift SDK equivalent
// Compile: vil compile --from swift --input 009-basic-credit-regulatory-slik/main.swift --release

let p = VilPipeline(name: "RegulatoryStreamPipeline", port: 3083)
p.sink(name: "regulatory_sink", port: 3083, path: "/regulatory-stream")
p.source(name: "regulatory_source", url: "http://localhost:18081/api/v1/credits/ndjson?count=1000", format: "json")
p.route(from: "sink.trigger_out", to: "source.trigger_in", mode: "LoanWrite")
p.route(from: "source.response_data_out", to: "sink.response_data_in", mode: "LoanWrite")
p.route(from: "source.response_ctrl_out", to: "sink.response_ctrl_in", mode: "Copy")
p.compile()
